src/cursor/cur_backup.c

```diff
@@ -144,11 +144,15 @@ __wt_curbackup_open(WT_SESSION_IMPL *session,
 
 	/*
 	 * Start the backup and fill in the cursor's list.  Acquire the schema
-	 * lock, we need a consistent view when creating a copy.
+	 * lock, we need a consistent view when creating a copy.  Further, if
+	 * backing up individual tables, we have to include indexes, which may
+	 * involve opening those indexes.  Acquire the table lock in write mode
+	 * for that case.
 	 */
 	WT_WITH_CHECKPOINT_LOCK(session,
 	    WT_WITH_SCHEMA_LOCK(session,
-		ret = __backup_start(session, cb, cfg)));
+		WT_WITH_TABLE_WRITE_LOCK(session,
```

Comment:
I'm a little surprised this is required - I thought backup went through {{__wt_meta_apply_all}} - avoiding the schema level operations. Does it do a schema apply as well?

Reply:
Yes, this is a "targeted backup" of named tables.

Reply:
Got it - now I see. It's unfortunate that the layering means we need to take that lock even for the more common full and incremental backup cases. I'll open a different ticket to address that.

Reply:
I've moved it in the latest commit so we only acquire that lock when doing targeted checkpoints.